### ISSUE TITLE
400 Bad Request  Oops, Saving menu failed: Failed to update main-menu.

### ISSUE THREAD

**Issue by iTerentius** [gantry/gantry5#1055]
I'm getting this when trying to Save the menu, with or without changes. I've deleted the existing menu and made new with one of one level with 4 pages.

This is WP 4.4 and Gantry5/Hydrogen 5.2.4

I haven't found a way to get around it. 

Oh, and on the front-end, the "Menu not found" is showing even though in the Gantry settings it is chosen.

**Comment by newkind** [comment]
Hi,

Can you please try if the error appears in Gantry 5.2.3 ?

**Comment by iTerentius** [comment]
Sure, let me swap it out.

**Comment by newkind** [comment]
Did this fixed it ?

**Comment by iTerentius** [comment]
This is probably a different issue related to permissions. However, this is the only thing I've installed that gives me issues. When I originally installed Hydrogen 5.2.4. We got this run-time exception. My server admin created the directory manually and gave it 777, but now on installing 5.2.3 I'm getting the same error: 

RuntimeException
HELP
Creating directory '/var/www/vhosts/labs.medicomhealth.com/wp-content/themes/g5_hydrogen/custom/config/default' failed on error mkdir(): Permission denied
Open: /var/www/vhosts/labs.medicomhealth.com/wp-content/plugins/gantry5/src/vendor/rockettheme/toolbox/File/src/File.php
        // Silence error for open_basedir; should fail in mkdir instead.
        if (!@is_dir($dir)) {
            $success = @mkdir($dir, 0777, true);
 
            if (!$success) {
                $error = error_get_last();
 
                throw new \RuntimeException("Creating directory '{$dir}' failed on error {$error['message']}");
            }
        }

**Comment by iTerentius** [comment]
We a non 'www-data' owner for our sites, but everything else works normally both on this install and the same way on our live install. Gantry is the only thing that has had a problem up to this point.

**Comment by newkind** [comment]
What about menus ?

**Comment by iTerentius** [comment]
I can't try 5.2.3 installed again until this issue is fixed. I can move this permission stuff to another issue.

**Comment by newkind** [comment]
777 is not safe, try setting wp-content/themes/hydrogen/custom a 775 permissions

**Comment by Joorik** [comment]
Hi! Have the same problem in Menu Editor: trying to save changes and always get 400 Bad reaquest: Oops, Saving menu failed: Failed to update mainmenu. I tried to do this on 5.2.3 and the last 5.2.4, but problem is the same. 
![menu editor](https://cloud.githubusercontent.com/assets/16286279/11776594/f6463cfc-a250-11e5-9f43-e84eced26f59.PNG)
Another problem - in the top menu of the final site displays only one item "Home" and other items from menu are missing. 
![main menu](https://cloud.githubusercontent.com/assets/16286279/11776602/fdc34182-a250-11e5-98b3-324c07275bb9.PNG)
Thank you in advance for your help.

**Comment by Tanver186** [comment]
I am also an issue with the menu rendering in WP4.4 BUT am not seeing any 404 error. It just fails to render correctly and instead only shows one item 'Research' even if there is no menu item named 'Research'

**Comment by w00fz** [comment]
To add to this. @Tanver186 shared with me his site and I was able to take a look.
There are 2 issues:

1. `400 Bad Request` which is due to the fact that the menu has no title set. On Tanver186 site I manually added the title to the menu and saved but on next refresh the title was gone again.
2. The menu he had was of 7 items and only 1 was rendering on frontend.

The only thing I noticed that could potentially be an issue is the name of the menu declared in Wordpress (`mnuMenu`) which on Menu Manager was displaying as `Mnumenu` and I am quite positive this gets saved as lowercase `mnumenu.yaml`. No idea if this is related but there's definitely something wrong with it.

**Comment by Tanver186** [comment]
@w00fz you may have noticed that even for the top menu named 'TopMenu' it displays 'Research' even though there is no menu item defined with that name, see this for details
http://prntscr.com/9dxnay

**Comment by iTerentius** [comment]
When it was working I also saw Joorik's single menu item as well.

This is only a problem on my corporate server with different permissions set up. A fresh install on Bluehost isn't an issue. Is there some permissions issues. I've already come up against this in another area. We have everything set to a user with somewhat limited permissions, but a group of www-data so that we have ftp ability under the user, and apache server permissions through www-data. Bluehost uses the root user for both the user and group I've noticed.

**Comment by Tanver186** [comment]
Is any body working on this issue? don't see it assigned to someone as yet.

**Comment by mahagr** [comment]
We need to find out a reliable way to reproduce the issue before we can locate and fix it. I have tried hard to reproduce this issue in my test environment, but I've not been lucky yet.

**Comment by iTerentius** [comment]
@mahagr, You might try changing the permissions of at least wp-content to 775 and or not having a full root access account for the user. Like my above message, I have the problem on our corporate server which has a user with limited privileges and www-data (apache's user) for the group - all 775. But not on a bluehost.com install where it is using a root account for both the user and group on the install.

**Comment by Tanver186** [comment]
@mahagr you can check with @w00fz who has access to my site where you can see the issue.

**Comment by DamirPecnik** [comment]
If you need access to broken website with this error @mahagr or @w00fz please look here:
http://www.rockettheme.com/forum/wordpress-theme-salient/252127-missing-menu-even-when-a-primary-is-selected

**Comment by gobbledygooch** [comment]
I believe I have this issue too. My site has 4 menus, 1 works fine.. the other 3 are showing one page (that isn't even in the menu). When you go into the menu that works fine it saves, the other 3 give the error:
400 Bad Request
Oops, Saving menu failed: Failed to update footer-links-menu.

I've taken content in and out of the menus, tried creating new menus with no effect.

Unfortunately I can't put it up somewhere to replicate. It is running on IIS. Nothing in the generic php log.

Let me know if I can answer any further questions.

**Comment by mahagr** [comment]
Please be free to jump into gitter chat at https://gitter.im/gantry/gantry5 and ping either @mahagr or @newkind in there.

We are trying to track down the issue today (European timezones).

**Comment by w00fz** [comment]
We understand this is an issue and we are trying to resolve it but none of us can reproduce this in our environments: we need your help.

Please upload a screenshot of the menu portion on frontend, a screenshot of your wordpress menu setup (Appearance -> Menu), a screenshot of the gantry menu manager (Hydrogen Theme -> Menu) and a zip of the menu folder that has been saved in `wp-content/themes/hydrogen/custom/config/menu`.

Thank you for your help.

**Comment by gobbledygooch** [comment]
I missed your message earlier. Unfortunately I don't have the files with me now but I will upload the info in about 8 hours.

**Comment by iTerentius** [comment]
Yeah, again - permissions - at least on my end. I don't have the /custom/config/menu/ directory (which would explain the error). There was a bunch of issues when installing the theme when it was trying to create the /custom directory structure. However we thought we got past it, but I see that it did not. Manually copying a working custom folder structure over (with menu), then just throws a bunch of errors by: wp-content/plugins/gantry5/src/vendor/rockettheme/toolbox/File/src/File.php as well. So it won't create nor will it work with manually copied folders and files.

Trying to create (activating gantry and timber):
![image](https://cloud.githubusercontent.com/assets/4072714/11873582/db526900-a4a0-11e5-85b0-346bd66b4a1e.png)

Reading manually placed custom folder with subfolders and files:
![image](https://cloud.githubusercontent.com/assets/4072714/11873508/83a3da72-a4a0-11e5-99a9-466741683805.png)

**Comment by iTerentius** [comment]
Oh, and we have SSH enabled. This prompt for user and pass comes up with every admin side update or install. This is probably part of the issue as well. I bet this is not getting passed along when creating/writing.

![image](https://cloud.githubusercontent.com/assets/4072714/11873710/a35bbb4a-a4a1-11e5-9236-48643089ef6f.png)

**Comment by mahagr** [comment]
Can you join our gitter chat and ping me? I really want to solve this issue for (all of) you, but nobody has given ssh access (for test site, not real one!) to me so I could track it down.

**Comment by mahagr** [comment]
Should be fixed now in latest CI build; please reopen if there are still issues.

**Comment by Coustoulin** [comment]
Ok 
I already chatted about this 
Thanks
